tac/governing-documents/project-lifecycle.md

```diff
@@ -142,3 +142,133 @@ state independent of its prior history.
+
+In addition, a project's lifecycle state should be evaluated during the quarterly updates if the TAC sees that a project is struggling to meet the requirements of its current state.
+
+(*Note*: as a rough hierarchy, the *Graduated* state represents the highest maturity, followed by *Incubated*, followed by *Dormant*, followed by *Archived*.)
+
+The following criteria MUST be considered for each project when making one of the aforementioned decisions:
+- Legal
+- Diversity
+- Releases
+- Testing and Q/A
+- Security
+- Structure
+- Maintenance
+- Production
+- Documentation
+
+## Description and Breakup of Criteria
+
+Below is a guide for evaluating each of these criteria, with break-ups into sub-criteria that can be directly evaluated using one of three methods:
+1. Checking their OpenSSF Scorecard scores
+2. Checking their LFX Insights Best Practice Scores (produced by CLOMonitor)
+3. Manual inspection of code repositories
+
+*Notes to keep in mind during evaluation*:
+- The term "close to" below implies that a high (or highest) score is not mandatory but highly desired. If a project falls short on some such criteria but is otherwise mature and well-functioning, the TAC may ignore those criteria while encouraging the project's maintainers to strive for a higher score.
+- Likewise, `SHOULD` and `SHOULD NOT` indicate that the stated criteria is subjective and not mandatory; if a project does not meet the given criteria, its status should be highlighted and debated during the [annual review](./project-annual-review.md) process.
+- A project may consist of multiple GitHub repositories. Several criteria below mention whether they must apply to at least one repository or to all repositories. Wherever this is not mentioned, the assumption should be that at least one repository must meet the given criterion. The project's maintainers shall have the right to designate one or more repositories for evaluation by the TAC whenever required, and these repositories may change from one cycle to the next.
+
+### Legal Criteria
+
+- (OpenSSF Score): *License*: MUST be 10
+
+### Diversity Criteria
+
+- (OpenSSF Score): *Contributors*: SHOULD be close to 10
+- (Manual): the `MAINTAINERS.md` files in the project's repositories MUST collectively indicate the involvement of more than one organization
+
+- Evaluation:
+  * To be a *Graduated* project, the number of organizations from which maintainers are drawn must be at least 3, and the OpenSSF Score SHOULD be 10.
+  * To be an *Incubated* project, the number of organizations from which maintainers are drawn must be at least 2.
+
+### Releases
+
+- (OpenSSF Score): *Packaging*: SHOULD be 10
+- (Manual): determined by inspecting the project's repositories, at least one of the repositories MUST have 2 or more releases in the past year
+
+### Testing and Q/A
+
+- (OpenSSF Score): *CI Tests*: SHOULD be 10
+
+### Security
+
+- (OpenSSF Score): *Dangerous Workflow*: MUST be 10
```

Comment:
Leaving a comment to acknowledge the discussion in previous TAC meetings regarding the subjectivity of this criteria. Given that the OpenSSF Scorecard system is relatively new to many projects, some may initially receive scores below the required threshold. In some cases, existing security practices may not be fully detected by the Scorecard, while others may require additional tooling. As long as project maintainers are actively working to improve their OpenSSF Scores, a lower score in certain areas shouldn’t necessarily be a reason to downgrade an otherwise successful project.

Reply:
I agree. We don't intend these guidelines to enforce automatic lifecycle state changes, but rather they should be treated as aspirational, and the TAC should guide and encourage the project's maintainers to fulfil the given criteria. Also, wherever we say `MUST`, we believe that the criteria can be fulfilled with not too much effort on the maintainers' part.